specification/eventgrid/Azure.Messaging.EventGrid/main.tsp

```diff
@@ -0,0 +1,294 @@
+    id: string;
+
+    @doc("Identifies the context in which an event happened. The combination of id and source must be unique for each distinct event.")
+    source: string;
+
+    @doc("Event data specific to the event type.")
+    data?: unknown;
+
+    @doc("Event data specific to the event type, encoded as a base64 string.")
+    data_base64?: bytes;
+    
+    @doc("Type of event related to the originating occurrence.")
+    type: string;
+
+    @doc("The time (in UTC) the event was generated, in RFC3339 format.")
+    time?: utcDateTime;
+
+    @doc("The version of the CloudEvents specification which the event uses.")
+    specversion: string;
+
+    @doc("Identifies the schema that data adheres to.")
+    dataschema?: string;
+
+    @doc("Content type of data value.")
+    datacontenttype?: string;
+
+    @doc("This describes the subject of the event in the context of the event producer (identified by source).")
+    subject?: string;
+  }
+
+  @doc("Properties of the Event Broker operation.")
+  model BrokerProperties {
+    @doc("The token used to lock the event.")
+    lockToken: string;
+
+    @doc("The attempt count for deliverying the event.")
+    deliveryAttemptCount: int32;
+  }
+
+  @doc("Receive operation details per Cloud Event.")
+  model ReceiveDetails {
+    @doc("The Event Broker details.")
+    brokerProperties: BrokerProperties;
+
+    @doc("Cloud Event details.")
+    event: CloudEvent;
+  }
+
+  @doc("Details of the Receive operation response.")
+  model ReceiveResult {
+    @doc("Array of receive responses, one per cloud event.")
+    value: ReceiveDetails[];
+  }
+   
+  @doc("Failed LockToken information.")
+  model FailedLockToken {
+    @doc("LockToken value")
+    lockToken: string;
+
+    @doc("Error code related to the token. Example of such error codes are BadToken: which indicates the Token is not formatted correctly, TokenLost: which indicates that token is not found, and InternalServerError: For any internal server errors.")
+    errorCode: string;
+
+    @doc("Description of the token error.")
+    errorDescription: string; 
+  }
+
+  @doc("The result of the Publish operation.")
+  model PublishResult {}
+
+  @doc("The result of the Acknowledge operation.")
+  model AcknowledgeResult {
+    @doc("Array of LockToken values for failed cloud events. Each LockToken includes the lock token value along with the related error information (namely, the error code and description).")
+    failedLockTokens: FailedLockToken[];
+
+    @doc("Array of lock tokens values for the successfully acknowledged cloud events.")
+    succeededLockTokens: string[];
+  }
+
+  @doc("The result of the Release operation.")
+  model ReleaseResult {
+    @doc("Array of LockToken values for failed cloud events. Each LockToken includes the lock token value along with the related error information (namely, the error code and description).")
+    failedLockTokens: FailedLockToken[];
```

Comment:
Release/Reject/Acknowledge will return 200 response even if a lock token has failed, and the error code for that lock token failure is placed under this FailedLockToken array with errorCode and errorDescription.

Reply:
In particular, even when there are only failures we get a 200. So there are three cases:
- all successes
- partial success/failure
- all failures

All three cases return 200.